src/reanimated2/isSharedValue.ts

```diff
@@ -1,7 +1,10 @@
 'use strict';
 import type { SharedValue } from './commonTypes';
 
-export function isSharedValue<T>(value: any): value is SharedValue<T> {
+export function isSharedValue<T = unknown>(
+  value: unknown
+): value is SharedValue<T> {
   'worklet';
-  return value?._isReanimatedSharedValue === true;
+  // We cannot use `in` operator here because `value` could be a HostObject and therefore we cast.
+  return (value as Record<string, unknown>)?._isReanimatedSharedValue === true;
```

Comment:
Since we export `isSharedValue` to the public API, should we extend the current implementation so that it supports arbitrary JS values? Currently `isSharedValue(42)` will fail.

btw why not `object` instead of `Record<string, unknown>`?

Reply:
With this implementation it works.
![Screenshot 2023-12-14 at 17 42 43](https://github.com/software-mansion/react-native-reanimated/assets/40713406/c2d931c0-683b-4345-9d41-c148e7a485c5)

Regarding the use of `Record<string,unknown>` - `object` is a type that doesn't have any properties so trying to access anything results in a TS error. Proper type to check for props is unfortunately `Record`.

Reply:
:like:

Reply:
Why don't we implement this function like this:
```jsx

export function isSharedValue<T>(value: unknown): value is SharedValue<T> {
  'worklet';
  return typeof value === 'object' &&
    value !== null &&
    '_isReanimatedSharedValue' in value
    ? value?._isReanimatedSharedValue === true
    : false;
}
```

Reply:
Longer code, but I'd prefer to avoid unnecessary type casts

Reply:
We can't use `in` operator because of HostObjects.